### ISSUE TITLE
3 bugs I have found

### ISSUE THREAD

**Issue by PoisonousSkely** [RedLime/SpeedRunIGT#46]
Hey so I have been playing SSGP pre 1.9 on version 1.8.9 with the timer mod and I have come across 3 issues.
1. In a lot of my runs the timer starts before I move even tho I have set it to ''first input''. I think it's because sometimes when I spawn in I am in the ground for a second and then I get pushed up so it thinks I moved but it doesn't always happen so I'm not too sure.
2. The timer doesn't account for portal lag. Sometimes when I enter the end portal or the portal to finish the game the game lags and it takes me a bit to actually go through which isn't supposed to count in your time afaik but the timer doesnt account for that and I lose some time because of it.
3. So when my game freezes the igt timer stops which is really nice. Sometimes when I kill the dragon my game freezes for a couple seconds and the timer stops, however the dragon death animation is still happening so I get a 2-3 sec timesave that I shouldnt get. For example I kill the dragon at 3:30 IGT, meaning the fountain should appear at 3:40 since the dragon death animation always takes 10 seconds. However my game freezes for 3 seconds and the timer stops so the fountain appears at 3:37 IGT instead according to the timer.

**Comment by RedLime** [comment]
what are you using speedrunigt version?

**Comment by PoisonousSkely** [comment]
10.1

**Comment by RedLime** [comment]
could you test with 10.2?

**Comment by PoisonousSkely** [comment]
nothing changed

**Comment by RedLime** [comment]
so could you upload video about this issue?

**Comment by PoisonousSkely** [comment]
You can see the 1st and 3rd issue I mentioned in this run (I will send something about the portal lag later)
https://www.youtube.com/watch?v=Atg1jNE_hxE
at 00:17 you can see my timer started even tho I didnt move (yes the setting was on first input) and at 3:21 you can see I killed the dragon at 2:48 but the fountain appeared at 2:55 instead of 2:58

**Comment by PoisonousSkely** [comment]
and here is an example of the igt timer not stopping on portal lag https://youtu.be/vM1wxWmFqIE

**Comment by RedLime** [comment]
```
1. In a lot of my runs the timer starts before I move even tho I have set it to ''first input''. I think it's because sometimes when I spawn in I am in the ground for a second and then I get pushed up so it thinks I moved but it doesn't always happen so I'm not too sure.
```
You probably enabled 'Legacy IGT Mode' option. disable it.

```
2. The timer doesn't account for portal lag. Sometimes when I enter the end portal or the portal to finish the game the game lags and it takes me a bit to actually go through which isn't supposed to count in your time afaik but the timer doesnt account for that and I lose some time because of it.
3. So when my game freezes the igt timer stops which is really nice. Sometimes when I kill the dragon my game freezes for a couple seconds and the timer stops, however the dragon death animation is still happening so I get a 2-3 sec timesave that I shouldnt get. For example I kill the dragon at 3:30 IGT, meaning the fountain should appear at 3:40 since the dragon death animation always takes 10 seconds. However my game freezes for 3 seconds and the timer stops so the fountain appears at 3:37 IGT instead according to the timer.
```
I CAN'T make any exceptions for these. because, as you know, timer is counting by client ticks. (for the minimize the gap between the play and timer displays and the performance differences)
however, minecraft's client world works a bit differently which causes this issue in Pre 1.13.
so if it does get the time correct from these issues, it must be check server side also, but this is too complicate and buggy.

yeah there's not much I can do about these issues. It's too risky, and it can break the timer or vanilla parity.
so the my recommend is just "retime". SpeedRunIGT provides a log for freezes like 3rd issue, you can use it to retime yourself or ask your moderator.

**Comment by PoisonousSkely** [comment]
ok ty

**Comment by PoisonousSkely** [comment]
Hey so about the first issue
It wasnt Legacy IGT Mode causing it as it was already off. I really dont know why it happens. Ig its ok if you cant fix it its just a bit annoying. It's literally rng if its gonna work properly or not every time I create a new world kek

**Comment by RedLime** [comment]
oh then can you send me the log files that saved when the the run is finished(you can use cheat) with occurred that issue?

**Comment by PoisonousSkely** [comment]
what file do you want me to send exactly?

**Comment by RedLime** [comment]
timer_igt.log

**Comment by PoisonousSkely** [comment]
[igt_timer.log](https://github.com/RedLime/SpeedRunIGT/files/8732976/igt_timer.log)
is this it?

**Comment by RedLime** [comment]
could you try again with 10.5?

**Comment by PoisonousSkely** [comment]
it still happens

**Comment by RedLime** [comment]
Fixed in 10.7.1
